kernel/nativelib.ml

```diff
@@ -27,15 +27,42 @@ let open_header = List.map mk_open open_header
 (* Directory where compiled files are stored *)
 let output_dir = ".coq-native"
 
-(* Extension of genereted ml files, stored for debugging purposes *)
+(* Extension of generated ml files, stored for debugging purposes *)
 let source_ext = ".native"
 
 let ( / ) = Filename.concat
 
-(* We have to delay evaluation of include_dirs because coqlib cannot be guessed
-until flags have been properly initialized *)
+(* Directory for temporary files for the conversion and normalisation
+   (as opposed to compiling the library itself, which uses [output_dir]). *)
+let my_temp_dir =
+  let make () =
+    let f = Filename.temp_file "Coq_native" "" in
+    Sys.remove f;
+    Unix.mkdir f 0o700;
+    f
```

Comment:
Instead of hacking this, you can do what the OCaml runtime does. It simply generates a random name using the function below. No need to access the disk and erase stuff.
```
let prng = lazy(Random.State.make_self_init ())

let temp_file_name temp_dir prefix suffix =
  let rnd = (Random.State.bits (Lazy.force prng)) land 0xFFFFFF in
  concat temp_dir (Printf.sprintf "%s%06x%s" prefix rnd suffix)
```

Reply:
(There is a loop trying 1000 times a random number in the generation but you can do the same.)

Reply:
@SkySkimmer ping, it would be nice for this to get into 8.11β1

Reply:
Changed